### PR TITLE
Checks reading is within the value type range when transforming

### LINKED ISSUE #88: The overflow in transform doesn't be handled
The result string should be set to "Overflow failed for device resource: ..." according to the requirement document, if the transform result is overflow the data type.

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #88 
Signed-off-by: Bruce <weichou1229@gmail.com>

### CODE REVIEW

File: internal/transformer/transformresult.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"context"
 	"encoding/binary"
 	"fmt"
+	"github.com/pkg/errors"
```
Comment (cloudxxx8) [inline internal/transformer/transformresult.go]: split up the import

File: go.mod [hunk 2]
```diff
@@ -20,6 +20,7 @@ require (
 	github.com/kr/logfmt v0.0.0-20140226030751-b84e30acd515
 	github.com/mitchellh/go-homedir v1.0.0
 	github.com/pelletier/go-toml v1.2.0
+	github.com/pkg/errors v0.8.1
```
Comment (cloudxxx8) [inline go.mod]: modify Attribution.txt if you add any new library

File: internal/transformer/transformresultchecker.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2018 IOTech Ltd
```
Comment (cloudxxx8) [inline internal/transformer/transformresultchecker.go]: 2019

File: internal/transformer/transformresultchecker_test.go [hunk 4]
```diff
@@ -0,0 +1,223 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2018 IOTech Ltd
```
Comment (cloudxxx8) [inline internal/transformer/transformresultchecker_test.go]: 2019

File: internal/transformer/overfloerror.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
```
Comment (jduranf) [inline internal/transformer/overfloerror.go]: File name should be `overflowerror.go`

Comment (weichou1229) [reply internal/transformer/overfloerror.go]: Renamed.

File: internal/transformer/transformresultchecker_test.go (hunk 4 above)
Comment (weichou1229) [reply internal/transformer/transformresultchecker_test.go]: Modified.

File: internal/transformer/transformresultchecker.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		}
+	case float32:
+		if math.Abs(transformed) >= math.SmallestNonzeroFloat32 && math.Abs(transformed) <= math.MaxFloat32 {
+			inRange = true
+		}
+	case float64:
+		if math.Abs(transformed) >= math.SmallestNonzeroFloat64 && math.Abs(transformed) <= math.MaxFloat64 {
+			inRange = true
+		}
+	}
```
Comment (cloudxxx8) [inline internal/transformer/transformresultchecker.go]: add default, and log error "data type %T doesn't support range checking", and return false

Comment (weichou1229) [reply internal/transformer/transformresultchecker.go]: Modified.

**Comment by cloudxxx8** [comment]
LGTM

**Comment by cloudxxx8** [comment]
recheck

**Comment by cloudxxx8** [comment]
recheck

**Pull request merged**